internal/pkg/manifest/interpolate.go

```diff
@@ -38,30 +40,29 @@ func newInterpolator(envVars predefinedEnvVar) *interpolator {
 }
 
 func (i *interpolator) substitute(s string) (string, error) {
-	matches := interpolatorRegexp.FindAllStringSubmatch(s, -1)
+	matches := interpolatorEnvVarRegExp.FindAllStringSubmatch(s, -1)
 	if len(matches) == 0 {
 		return "", nil
 	}
 	replaced := s
 	for _, match := range matches {
 		// https://pkg.go.dev/regexp#Regexp.FindAllStringSubmatch
 		key := match[1]
-		subTextRegexp := regexp.MustCompile(fmt.Sprintf(`\${%s}`, key))
+		currSegmentRegex := regexp.MustCompile(fmt.Sprintf(`\${%s}`, key))
 		predefinedVal, isPredefined := i.predefinedEnvVars[key]
-		if isPredefined {
-			replaced = subTextRegexp.ReplaceAllString(replaced, predefinedVal)
-		}
 		osVal := os.Getenv(key)
-		if osVal == "" {
-			if isPredefined {
-				continue
-			}
-			return "", fmt.Errorf(`environment variable "%s" is not defined`, key)
+		if isPredefined && osVal != "" {
+			return "", fmt.Errorf(`predefined environment variable "%s" cannot be overridden with "%s"`, key, osVal)
```

Comment:
```suggestion
			return "", fmt.Errorf(`predefined environment variable "%s" cannot be overridden by OS environment variable with the same name`)
```
We probably shouldn't include their os val in error message🤔 just to be safe.

I am also wondering if we need to error out in this case. Maybe we can just display a message telling them, well we detected that you have an env var with name colliding with our predefined var. We are using our predefined value instead of yours - so that people do not have to unset their OS env var in order to get this to work.

Reply:
make sense!